src/types.ts

```diff
@@ -922,16 +927,20 @@ export class Signature {
     type.signatureReference = this;
 
     let signatureTypes = program.uniqueSignatures;
-    let length = signatureTypes.length;
-    for (let i = 0; i < length; i++) {
-      let compare = unchecked(signatureTypes[i]);
-      if (this.equals(compare)) {
-        this.id = compare.id;
-        return this;
+    let compareString = this.toString();
+    if (signatureTypes.has(compareString)) {
+      let existing = signatureTypes.get(compareString)!;
```

Comment:
Nit: When compiling with `asc`, this line yields a "Value is never null" kind of warning. Is somewhat a design wart resulting from JS being typed with `| undefined` here. A way around for now is
```suggestion
      let existing = changetype<Signature>(signatureTypes.get(compareString));
```